src/main/cli/cli.c

```diff
@@ -6783,31 +6790,65 @@ void cliProcess(void)
 
     while (serialRxBytesWaiting(cliPort)) {
         uint8_t c = serialRead(cliPort);
+        if (cliInteractive) {
+            processCharacterInteractive(c);
+        } else {
+            // handle terminating flow control character
+            if (c == 0x3) { // CTRL-C
+                cliWrite(0x3); // send end of text, terminating flow control
+                cliExit(false);
+                return;
+            }
+            processCharacter(c);
+        }
+    }
+}
+
+static void cliExit(const bool reboot)
+{
+    cliWriterFlush();
+    waitForSerialPortToFinishTransmitting(cliPort);
+    *cliBuffer = '\0';
+    bufferIndex = 0;
+    cliMode = false;
+    cliInteractive = false;
+    // incase a motor was left running during motortest, clear it here
+    mixerResetDisarmedMotors();
 
-        processCharacterInteractive(c);
+    if (reboot) {
+        cliReboot();
     }
 }
 
-void cliEnter(serialPort_t *serialPort)
+void cliEnter(serialPort_t *serialPort, bool interactive)
 {
     cliMode = true;
+    cliInteractive = interactive;
     cliPort = serialPort;
-    setPrintfSerialPort(cliPort);
+
+    if (interactive) {
```

Comment:
Why interactive only?

Reply:
Because you don't want debug output in the non interactive command output